File: lv1-library-loan/csharp/LibraryLoan.Domain/Members/MemberErrors.cs
using LibraryLoan.Results;

namespace LibraryLoan.Domain.Members;

public static class MemberErrors
{
    public static Error HasOverdueLoan(MemberId memberId) => new(
        "Member.HasOverdueLoan",
        $"会員(ID: {memberId})は延滞中の貸出記録があるため、新しく本を借りることができません。");
}

File: lv1-library-loan/typescript/src/domain/members/member-errors.ts
import { ResultError } from "../../results/result-error.js";
import type { MemberId } from "./member-id.js";

export const MemberErrors = {
  hasOverdueLoan: (memberId: MemberId): ResultError =>
    new ResultError(
      "Member.HasOverdueLoan",
      `会員(ID: ${memberId})は延滞中の貸出記録があるため、新しく本を借りることができません。`,
    ),
};
